### PR TITLE
fix: Fix node graph telemetry with default values

### PR DESCRIPTION
**Pull request by mutdmour**
## Summary
- Default values for nodes don't get set correctly. Stop resolving them directly and use helper for this
- Stop swallowing errors for this function

## Related tickets and issues
https://linear.app/n8n/issue/ADO-376/possible-bug-in-resourceoperation-analytics

## Review / Merge checklist
- [X] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [X] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

File: packages/workflow/src/TelemetryHelpers.ts [hunk 1]
```diff
@@ -129,106 +107,133 @@ export function generateNodesGraph(
 	const nodeNameAndIndex: INodeNameIndex = {};
 	const webhookNodeNames: string[] = [];
 
-	try {
```
Comment (mutdmour) [inline packages/workflow/src/TelemetryHelpers.ts:132]: removed try/catch that swallowed all errors

File: packages/workflow/src/TelemetryHelpers.ts [hunk 2]
```diff
@@ -129,106 +107,133 @@ export function generateNodesGraph(
...
+						});
 					}
-				});
-
-				nodeItem.operation = node.parameters.operation?.toString() ?? nodeItem.operation;
-				nodeItem.resource = node.parameters.resource?.toString() ?? nodeItem.resource;
-				nodeItem.mode = node.parameters.mode?.toString() ?? nodeItem.mode;
+				}
+			} catch (e: unknown) {
+				if (!(e instanceof Error && e.message.includes('Unrecognized node type'))) {
```
Comment (mutdmour) [inline packages/workflow/src/TelemetryHelpers.ts:211]: not perfect. will follow up on this PR to try and standarize this somehow https://github.com/n8n-io/n8n/pull/8264

File: packages/workflow/src/TelemetryHelpers.ts [hunk 3]
```diff
@@ -129,106 +107,133 @@ export function generateNodesGraph(
...
-
-				nodeType?.description?.properties?.forEach((property) => {
-					if (
-						property.name === 'operation' ||
-						property.name === 'resource' ||
-						property.name === 'mode'
-					) {
-						nodeItem[property.name] = property.default ? property.default.toString() : undefined;
+				if (nodeType) {
+					const nodeParameters = getNodeParameters(
```
Comment (mutdmour) [inline packages/workflow/src/TelemetryHelpers.ts:189]: main fix is here to reuse this logic to resolve parameters.

File: packages/workflow/src/TelemetryHelpers.ts [hunk 4]
```diff
@@ -129,106 +107,133 @@ export function generateNodesGraph(
...
 			}
-			nodesGraph.nodes[`${index}`] = nodeItem;
-			nodeNameAndIndex[node.name] = index.toString();
-		});
+		}
 
-		const getGraphConnectionItem = (startNode: string, connectionItem: IConnection) => {
-			return { start: nodeNameAndIndex[startNode], end: nodeNameAndIndex[connectionItem.node] };
-		};
+		nodesGraph.nodes[`${index}`] = nodeItem;
```
Comment (tomi) [inline packages/workflow/src/TelemetryHelpers.ts:217]: nit: I believe the conversion to string in `nodesGraph.nodes[`${index}`]` is unnecessary. Or if it's a TS error could use `index.toString()` in both cases

File: packages/workflow/src/TelemetryHelpers.ts [hunk 5]
```diff
@@ -129,106 +107,133 @@ export function generateNodesGraph(
...
+		nodeNameAndIndex[node.name] = index.toString();
```
Comment (tomi) [inline packages/workflow/src/TelemetryHelpers.ts:218]: Not the greates of names. `nodeIndexByName` would be more descriptive. But yeah probably out of scope of this PR. The function is really long anyways could use some ❤️.

**Review by tomi — DISMISSED** [review]
LGTM 💯 Really nice job adding those tests 😍

**Review by tomi — DISMISSED** [review]

**Review by tomi — DISMISSED** [review]

**Pull request merged**